flow/Platform.actor.cpp

```diff
@@ -3635,7 +3640,10 @@ void crashHandler(int sig) {
 
 	StreamCipherKey::cleanup();
 	StreamCipher::cleanup();
-	BlobCipherKeyCache::cleanup();
+
+	for (auto& f : g_crashHandlerCallbacks) {
+		f();
+	}
```

Comment:
Is it necessary to do this cleanup on a crash? We can't handle all abrupt termination paths here. It also may not be safe to do much in this function, and if this fails it might prevent some of the subsequent data logging that we rely on.

Reply:
The cache being cleared stores EncryptionKeys, one of the limitations of the design currently is, that a process dump isn't encrypted and this code attempts to clean up cache as part of the process dump. The operation clears underlying `std::unordered_map` so usually shouldn't fail, unless, the machine is observing hardware failures (faulty memory)

Further, BlobCipher leveraged existing Platform cleanup code already used by `StreamCipher` and others, so assuming not introducing any new workflow. 

Long term plan is to introduce an `arena` that leverage `madvise` `MADV_DONTDUMP` flag to avoid memory region containing sensitive information (Encryption keys in this context) from getting dumped as part of a process core dump.

Reply:
I'm not sure how good the contract the signal handler has is for security purposes, but assuming it doesn't otherwise cause problems we may not be comprehensively handling all the relevant signals. It's at least worth a check.

The other thing I would advise is to add some documentation to this callback mechanism to actually explain what the contract is and probably to discourage its use in most cases. Even if we are comfortable that the current changes won't break the crash logging, by adding this mechanism we make it easier for someone else to make that mistake in the future.

Reply:
Good point, I think documenting the contract PLUS adding a `TODO` explaining the BlobCipher callback registration as a `stop-gap` solution is definitely very useful.

Reply:
I'm only trying to keep existing logic here. IMO the cleanup only avoid leaking cipher keys, but not user data, into core dumps, so it doesn't quite achieve what we want. I incline to not doing this cleanup, but push for disabling core dumps altogether. @sfc-gh-ahusain what do you think?

Reply:
> but push for disabling core dumps altogether

IMO, core dumps are necessary for field triaging, especially for issues such as: memory leak, logic bugs etc. I would suggest not making this change and let the code be in its state. It feels better to take cleanup as a separate patch. Thoughts?

Reply:
definitely.

Reply:
Added more comments and TODO.